src/form/Calendar/Calendar.tsx

```diff
@@ -172,21 +191,61 @@ export const Calendar: FC<CalendarProps> = ({
   }, [onViewChange, view]);
 
   const dateChangeHandler = useCallback(
-    (date: Date) => {
+    (newDate: Date) => {
+      let finalDate = newDate;
+      if (showTime && value) {
```

Comment:
If this gets much more complicated, we should extract it to a pure function and add some tests.